### ISSUE TITLE
Update all keys from keyserver crash the app

### ISSUE THREAD

**Issue by jdstrydom** [open-keychain/open-keychain#2345]
OpenKeychain 5.1 (newest) resolved an issue on Android 5.1.1 for me when K-9 mail was unable to select a relevant key causing OpenKeychain to crash.

Unfortunately the new version introduces a new issue where OpenKeychain crashes when "Update all keys" is selected from the menu

## Expected Behavior
Keys should be updated to/from the keyserver

## Current Behavior
OpenKeychain crashes

## Possible Solution
Reverting to version 5.0 I had available, solves the update all keys problem

## Steps to Reproduce (for bugs)

1. Open OpenKeychain
2. Select the 3-dot menu in the top right-hand corner 
3. Click on "Update all keys"
4. App crash

## Your Environment
[openkeychain.txt](https://github.com/open-keychain/open-keychain/files/2098145/openkeychain.txt)

* Android Version: 5.1.1
* Device Model: Redmi Note 2
* OpenKeychain Version: 5.1
* From Google Play or F-Droid?: F-Droid

**Comment by jdstrydom** [comment]
Will try with the new OpenKeyChain 5.1.1 released today.

**Comment by Valodim** [comment]
we fixed one crash in that area, so this might be fixed. please report back if it works with 5.1.1 :)

**Comment by jdstrydom** [comment]
I edited my comment after installing OpenKeychain 5.1.1, and attached logcat.  The crash still happens.

**Comment by gtalge** [comment]
I am having the same problem. Unfortunately google play does not have a way to downgrade. The previous version worked without issue.

**Comment by gtalge** [comment]
Hi,

It immediately crashes as soon as I tap "Update all keys" or when I try to update any key at all.

It all worked fine before the recent update.

I have an LG Treasure with
Android 5.1.1 Kernel 3.10.49

Gordon

On June 13, 2018 4:24:23 PM PDT, Vincent Breitmoser <notifications@github.com> wrote:
>When you click "update all keys", does it crash right away or does it
>update some keys and then crash? I tried on a couple of test devices
>and it didn't crash on any of them, so it might be some property of
>your keys or the way we save them that causes the crash.
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/open-keychain/open-keychain/issues/2345#issuecomment-397118981

-- 
Deo Vindice

PGP: AA32DFCC (2023/07/14)

**Comment by gtalge** [comment]
I don't know if this helps to fix the Openkeychain crashing, but when
you go to update all keys, for a split second, the screen appears with a line with 0% at the end.

Kind of like this:

-------------------------------------------------------- 0%

This only lasts for maybe 1/4 of a second or less, just a flash.

Then it goes to 

"Unfortunately, Openkeychain has stopped."

I have sent in reports with logs to google play.

**Comment by Valodim** [comment]
Thanks for your detailed report! We finally received a stack trace that also shows the causing exceptions in the play console. Those were missing in your log file above, that's why this was hard to debug :)

```
java.lang.RuntimeException: 
  at org.sufficientlysecure.keychain.operations.ImportOperation.multiThreadedKeyImport (ImportOperation.java:573)
  at org.sufficientlysecure.keychain.operations.ImportOperation.execute (ImportOperation.java:513)
  at org.sufficientlysecure.keychain.operations.ImportOperation.execute (ImportOperation.java:87)
  at org.sufficientlysecure.keychain.service.KeychainService$1.run (KeychainService.java:148)
  at java.lang.Thread.run (Thread.java:818)
Caused by: java.util.concurrent.ExecutionException: 
  at java.util.concurrent.FutureTask.report (FutureTask.java:94)
  at java.util.concurrent.FutureTask.get (FutureTask.java:164)
  at org.sufficientlysecure.keychain.operations.ImportOperation.multiThreadedKeyImport (ImportOperation.java:564)
Caused by: java.lang.NullPointerException: 
  at org.sufficientlysecure.keychain.keyimport.HkpKeyserverClient.getResponseBodyAsUtf8 (HkpKeyserverClient.java:422)
  at org.sufficientlysecure.keychain.keyimport.HkpKeyserverClient.query (HkpKeyserverClient.java:400)
  at org.sufficientlysecure.keychain.keyimport.HkpKeyserverClient.get (HkpKeyserverClient.java:315)
  at org.sufficientlysecure.keychain.operations.ImportOperation.fetchKeyFromKeyserver (ImportOperation.java:395)
  at org.sufficientlysecure.keychain.operations.ImportOperation.fetchKeyFromInternet (ImportOperation.java:348)
  at org.sufficientlysecure.keychain.operations.ImportOperation.serialKeyRingImport (ImportOperation.java:195)
  at org.sufficientlysecure.keychain.operations.ImportOperation.access$000 (ImportOperation.java:87)
  at org.sufficientlysecure.keychain.operations.ImportOperation$1.call (ImportOperation.java:553)
  at org.sufficientlysecure.keychain.operations.ImportOperation$1.call (ImportOperation.java:541)
  at java.util.concurrent.FutureTask.run (FutureTask.java:237)
  at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:423)
  at java.util.concurrent.FutureTask.run (FutureTask.java:237)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1113)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:588)
```

**Comment by gtalge** [comment]
Downloaded openkeychain 5.0.2 from F-Droid after uninstalling 5.1.1 from Google play store.

The key update issue went away with the down grade, but get this:

I did something to the settings of K-9 mail and it wouldn't work with openkeychain anymore. 

The entry under account setting for choosing a key would crash openkeychain! I couldn't set up what secret key to use.

K-9 Mail was installed from google play store.

The solution was to uninstall 5.0.2 and install 5.1.1 from playstore, then import all keys and setup K-9 to work with openkeychain, then force stop K-9 mail and uninstall 5.1.1 from google play store, then reinstall 5.0.2 from F-Droid and re-import all the keys,
then open and restart K-9 Mail!

Now it works ok with updating keys and with K-9.

Really strange, but worked.

Gordon



On June 14, 2018 4:53:11 AM PDT, Vincent Breitmoser <notifications@github.com> wrote:
>Closed #2345 via 996856ddde041b95455aff555d61020f66b618ba.
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/open-keychain/open-keychain/issues/2345#event-1680960920

---
Deo Vindice

PGP : AA32DFCC (2023/07/14)
